momento/leaderboard_data_client.go

```diff
@@ -0,0 +1,209 @@
+package momento
+
+import (
+	"context"
+	"time"
+
+	"github.com/momentohq/client-sdk-go/internal/grpcmanagers"
+	"github.com/momentohq/client-sdk-go/internal/models"
+	"github.com/momentohq/client-sdk-go/internal/momentoerrors"
+	pb "github.com/momentohq/client-sdk-go/internal/protos"
+)
+
+type LeaderboardDataClient interface {
+	Close() momentoerrors.MomentoSvcErr
+	Delete(ctx context.Context, request *LeaderboardInternalDeleteRequest) momentoerrors.MomentoSvcErr
+	FetchByRank(ctx context.Context, request *LeaderboardInternalFetchByRankRequest) ([]*pb.XRankedElement, momentoerrors.MomentoSvcErr)
+	FetchByScore(ctx context.Context, request *LeaderboardInternalFetchByScoreRequest) ([]*pb.XRankedElement, momentoerrors.MomentoSvcErr)
+	GetRank(ctx context.Context, request *LeaderboardInternalGetRankRequest) ([]*pb.XRankedElement, momentoerrors.MomentoSvcErr)
+	Length(ctx context.Context, request *LeaderboardInternalLengthRequest) (uint32, momentoerrors.MomentoSvcErr)
+	RemoveElements(ctx context.Context, request *LeaderboardInternalRemoveElementsRequest) momentoerrors.MomentoSvcErr
+	Upsert(ctx context.Context, request *LeaderboardInternalUpsertRequest) momentoerrors.MomentoSvcErr
+}
+
+type leaderboardDataClient struct {
+	requestTimeout         time.Duration
+	leaderboardGrpcManager *grpcmanagers.LeaderboardGrpcManager
+	leaderboardClient      pb.LeaderboardClient
+}
+
+func NewLeaderboardDataClient(request *models.LeaderboardClientRequest) (*leaderboardDataClient, momentoerrors.MomentoSvcErr) {
```

Comment:
```suggestion
func newLeaderboardDataClient(request *models.LeaderboardClientRequest) (*leaderboardDataClient, momentoerrors.MomentoSvcErr) {
```
We don't want this to be public.